homeworks/dobrikbartaeva/4_exceptions_logging/to_do.py

```diff
@@ -0,0 +1,105 @@
+import os
+import logging
+
+to_do_list_path = "homeworks/dobrikbartaeva/4_exceptions_logging/to_do_list.txt"
+
+def setup_logger(logger_name, log_file="homeworks/dobrikbartaeva/4_exceptions_logging/app.log",level=logging.INFO):
+    logger = logging.getLogger(logger_name)                                         #Call getlogger function from logging module
+    file_handler = logging.FileHandler(log_file)                                    #Where we write the log messages
+    stream_handler = logging.StreamHandler()  
+                                            #Where we write the log messages
+    #Format of the messages:
+    formatter = logging.Formatter('%(asctime)s - %(levelname)s - %(message)s')
+
+    #Assign formatter to the handler:
+    file_handler.setFormatter(formatter)                                            
+    stream_handler.setFormatter(formatter)   
+
+    #Assign the level of the message to the handler 
+    file_handler.setLevel(logging.DEBUG)                                            
+    stream_handler.setLevel(logging.DEBUG)  
+
+    #Add handler to the logger:
+    logger.addHandler(file_handler)                                                 
+    logger.addHandler(stream_handler)                                               
+    logger.setLevel(level)                                                          
+    return logger
+
+logger = setup_logger("to_do_logger")
+
+with open(to_do_list_path, "w") as file:
+    file.write("")
+
+def add_task(task_input):
+    with open(to_do_list_path, "a") as file:
+        file.write(f"{task_input}\n")
+        print(f"You added: {task_input}")
+
+def view_tasks():
+    with open(to_do_list_path, "r") as file:
+        lines = file.readlines()
+    if not lines:
+        print("To do list is empty.")
+    else:
+        print("To do list:")   
+        for id, line in enumerate(lines,1):
+            print(f"{id}. - {line.strip()}")
+
+def remove_task(task_input):
```

Comment:
Ez a function egy kicsit lehetne effektívebb, mert így egyesével írod vissza a file-ba a feladatokat, és ahogy nő a file mérete ez egyre több időbe fog telni:

```
def remove_task(task_input):
    with open(to_do_list_path, "r") as file:
        lines = file.readlines()

    new_lines = [line for line in lines if line.strip() != task_input]
    if len(new_lines) == len(lines):
        print(f"This task is not on your to do list: {task_input}")
    else:
        with open(to_do_list_path, "w") as file:
            file.writelines(new_lines)
        print(f"You removed: {task_input}")
```

Itt egy lehetséges megoldás, beolvasod az összes sort egy listába, majd egy list comprehension-nel megtartod azokat a sorokat amik nem tartalmazzák a megadott taskot (akár többször is ugyanazt).
Ezután csak visszaírod az egész listát úgy ahogy van, egyszerre.